README.md

```diff
@@ -0,0 +1,153 @@
+# Relativity
+
+[![CI Status](https://travis-ci.org/dfed/Relativity.svg?branch=master)](https://travis-ci.org/dfed/Relativity)
+[![Carthage Compatibility](https://img.shields.io/badge/carthage-✓-e2c245.svg)](https://github.com/Carthage/Carthage/)
+[![Version](https://img.shields.io/cocoapods/v/Relativity.svg)](http://cocoadocs.org/docsets/Relativity)
+[![License](https://img.shields.io/cocoapods/l/Relativity.svg)](http://cocoadocs.org/docsets/Relativity)
+[![Platform](https://img.shields.io/cocoapods/p/Relativity.svg)](http://cocoadocs.org/docsets/Relativity)
+
+A programmatic layout engine and DSL that provides an alternative to Auto Layout.
+
+## Installation
+
+### CocoaPods
+
+To install Relativity in your iOS project with [CocoaPods](http://cocoapods.org), add the following to your `Podfile`:
+
+```
+platform :ios, '8.0'
+pod 'Relativity', '~> 0.8.0'
+```
+
+### Carthage
+
+To install Relativity in your iOS project with [Carthage](https://github.com/Carthage/Carthage), add the following to your `Cartfile`:
+
+```ogdl
+github "dfed/Relativity"
+```
+
+Run `carthage` to build the framework and drag the built `Relativity.framework` into your Xcode project.
+
+### Swift Package Manager
+
+To install Relativity in your iOS project with [Swift Package Manager](https://github.com/apple/swift-package-manager), the following definition can be added to the dependencies of your `Project`:
+
+```swift
+  .Package(url: "https://github.com/dfed/Relativity.git", majorVersion: 0, minor: 8),
+```
+
+### Submodules
+
+To use git submodules, checkout the submodule with `git submodule add git@github.com:dfed/Relativity.git`, drag Relativity.xcodeproj to your project, and add Relativity as a build dependency.
+
+## Usage
+
+### Programmatic Layout
+
+There are four basic steps to laying out views programmatically. In `init` or `viewDidLoad`:
+1. Create your views.
+2. Create your view hierarchy with `addSubview()`.
+Then in `layoutSubviews` for a `UIView` or `viewDidLayoutSubviews` for a `UIViewController`:
+3. Size your views.
+4. Position your views.
+
+Note that with the above steps your UI will animate nicely during screen rotations and status bar size changes.
+
+### Anchoring UIViews
+
+Positioning views are done by anchoring a corner or edge of a view to another view’s corner or edge, plus an offset. A `-->` moves the view on the left side to the position described on the right side. A `<--` moves the view on the right side to the position described on the left side. You can also explicitly align views using the `align(to:xOffset:yOffset)` method in [ViewPosition.swift](Sources/ViewPosition.swift#L105).
+
+#### Examples
+
+To align view `a` to be 10 points to the left of view `b`:
+
+```swift
+  a.rightCenter --> 10.horizontalOffset + b.leftCenter
```

Comment:
This syntax is confusing to me. Two suggestions, that I think would work:
1) Use `10` instead of `10.horizontalOffset`. I believe you can do this by creating custom operators for things like `+(CGFloat, ViewPosition) -> ViewPosition`
2) Use the `=` operator instead of `-->`. You can override the operator by overriding the `.rightCenter` setter, and the `ViewPosition` object that gets assigned can contain all of the necessary info about other views and offsets.

This gets you to (what I consider) a more clear and natural syntax of something like `b.leftCenter = a.rightCenter + 10`